core/domain/html_cleaner.py

```diff
@@ -83,3 +84,83 @@ def clean(user_submitted_html):
 def strip_html_tags(html):
     """Strips all HTML markup from an HTML string."""
     return bleach.clean(html, tags=[], attributes={}, strip=True)
+
+
+def textangular_to_ckeditor(content):
+    '''Converts rte content from textAngular format to CkEditor format.'''
+    # Wrapping div serves as a parent to everything else.
+    content = '<div>' + content + '</div>'
+    soup = BeautifulSoup(content, 'html.parser')
+
+    # Move block components out of <p>,<b>, and <i>
+    invalid_parents = ['p', 'b', 'i']
+    components_names = ['image', 'collapsible', 'tabs']
+    tag_names = ['oppia-noninteractive-' + component_name
+                 for component_name in components_names]
+    components = soup.find_all(tag_names)
+    for component in components:
+        while component.parent.name in invalid_parents:
+            parent_name = component.parent.name
+            previous_siblings = soup.new_tag(parent_name)
+            for prev_sibling in component.previous_siblings:
+                previous_siblings.append(prev_sibling)
+            next_siblings = soup.new_tag(parent_name)
+            for next_sibling in component.next_siblings:
+                next_siblings.append(next_sibling)
+            if previous_siblings.contents:
+                component.parent.insert_before(previous_siblings)
+            if next_siblings.contents:
+                component.parent.insert_after(next_siblings)
+            component.parent.replace_with(component)
+
+    # Bold tag conversion: 'b' -> 'strong'.
+    for bold in soup.find_all('b'):
+        bold.name = 'strong'
+
+    # Italic tag conversion: 'i' -> 'em'.
+    for italic in soup.find_all('i'):
+        italic.name = 'em'
+
+    # Replace all spans with whatever is inside them.
+    for span in soup.find_all('span'):
+        span.unwrap()
+
+    for br in soup.find_all('br'):
+        if br.contents:
+            # This happens in <pre> blocks, we want to replace br
+            # with newlines.
+            br.insert_before('\n')
+            br.unwrap()
+        else:
+            # In all other cases, just replace with whitespace.
+            br.replace_with(u'\xa0')
```

Comment:
Does this get converted to `&nbsp;` in the html? Is this what we get when typing a space in the RTE?

Are spaces always represented as `&nbsp;`? If not, when are they represented as such, and when are they represented as a space character (' ' )?